tests/unit/test_timeaverage.py

```diff
@@ -0,0 +1,263 @@
+    )  # Approximately two minutes in days
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("0.020833") == "30min"
+    )  # Approximately half an hour in minutes
+
+
+def test__frequency_from_approx_interval_second():
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("0.00001157") == "s"
+    )  # Approximately one second in days
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("0.00002314") == "2s"
+    )  # Approximately two seconds in days
+    assert not (
+        pymorize.timeaverage._frequency_from_approx_interval("0.000694") == "60s"
+    )  # Approximately one minute in seconds, should give back min, since it can round up.
+
+
+def test__frequency_from_approx_interval_millisecond():
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("1.1574e-8") == "ms"
+    )  # Approximately one millisecond in days
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("2.3148e-8") == "2ms"
+    )  # Approximately two milliseconds in days
+
+
+def test__invalid_interval():
+    with pytest.raises(ValueError):
+        pymorize.timeaverage._frequency_from_approx_interval("not_a_number")
+
+
+def test__compute_file_timespan_single_chunk():
+    # Create a DataArray with a single chunk
+    time = pd.date_range("2000-01-01", periods=10, freq="D")
+    data = xr.DataArray(np.random.rand(10), dims="time", coords={"time": time})
+    data = data.chunk({"time": 5})  # Single chunk
+
+    assert pymorize.timeaverage._compute_file_timespan(data) == 4
+
+
+def test__compute_file_timespan_multiple_chunks():
+    # Create a DataArray with multiple chunks
+    time = pd.date_range("2000-01-01", periods=30, freq="D")
+    data = xr.DataArray(np.random.rand(30), dims="time", coords={"time": time})
+    data = data.chunk({"time": 10})
+
+    assert pymorize.timeaverage._compute_file_timespan(data) == 9
+
+
+def test__compute_file_timespan_empty_time_dimension():
+    # DataArray with an empty time dimension
+    data = xr.DataArray(np.array([]), dims="time", coords={"time": []})
+    with pytest.raises(ValueError, match="no time values in this chunk"):
+        pymorize.timeaverage._compute_file_timespan(data)
+
+
+def test__compute_file_timespan_missing_time_dimension():
+    # DataArray without a time dimension
+    data = xr.DataArray(np.random.rand(10), dims="x")
+    with pytest.raises(ValueError, match="missing the 'time' dimension"):
+        pymorize.timeaverage._compute_file_timespan(data)
+
+
+def test__compute_file_timespan_non_sequential_time():
+    # DataArray with non-sequential time points
+    time = pd.to_datetime(["2000-01-01", "2000-01-05", "2000-01-20", "2000-01-30"])
+    data = xr.DataArray(np.random.rand(4), dims="time", coords={"time": time})
+    data = data.chunk({"time": 2})
+
+    # FIXME: I'm not sure how to define this test for correctness...
+
```

Comment:
This feature is not implemented in time averaging. This will turn out to be a expensive operation on large datasets with reshuffling data. Should this be handled in `gather_inputs.InputFileCollection` instead?

Reply:
I would like to leave this test in there for now (note that it doesn't actually test anything, and if the need arises for such a use case, we can define the actual behaviour then.